gnat2goto/ireps/ireps-follow_irep.adb

```diff
@@ -5,17 +5,31 @@ function Follow_Irep (L : Irep_List;
 is
    The_List : Irep_List_Node;
    Ptr      : Internal_Irep_List;
+   Arr      : constant Irep_List := New_List;
+   Modified : Boolean := False;
 begin
-   return Arr : constant Irep_List := New_List do
-      if L /= 0 then
-         The_List := Irep_List_Table.Table (To_Internal_List (L));
-         Ptr      := To_Internal_List (Irep_List (The_List.A));
-         while Ptr /= 0 loop
-            Append (Arr,
-                    Follow_Irep (Irep (Irep_List_Table.Table (Ptr).A),
-                                 Follow_Symbol));
+   if L /= 0 then
+      The_List := Irep_List_Table.Table (To_Internal_List (L));
+      Ptr      := To_Internal_List (Irep_List (The_List.A));
+      while Ptr /= 0 loop
+         declare
+            Orig_Irep : constant Irep := Irep (Irep_List_Table.Table (Ptr).A);
+            New_Irep  : constant Irep := Follow_Irep (Orig_Irep,
+                                                      Follow_Symbol);
+         begin
+            if Orig_Irep /= New_Irep then
```

Comment:
You are right about the first part: I should be enlisting `Orig_Irep` in case it was not modified. Wrt. the second part, I modified the code from `ireps-to_json.adb`, which doesn't mean it's the right way.